kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/client/DefaultParquetHandler.java

```diff
@@ -100,4 +106,62 @@ public CloseableIterator<DataFileStatus> writeParquetFiles(
             new ParquetFileWriter(hadoopConf, new Path(directoryPath), maxFileSize, statsColumns);
         return batchWriter.write(dataIter);
     }
+
+    /**
+     * Makes use of {@link LogStore} implementations in `delta-storage` to atomically write the data
+     * to a file depending upon the destination filesystem.
+     *
+     * @param filePath Fully qualified destination file path
+     * @param data     Iterator of {@link FilteredColumnarBatch}
+     * @throws IOException
+     */
+    @Override
+    public void writeParquetFileAtomically(
+            String filePath,
+            CloseableIterator<FilteredColumnarBatch> data) throws IOException {
+        Path targetPath = new Path(filePath);
+        LogStore logStore =
+                LogStoreProvider.getLogStore(hadoopConf, targetPath.toUri().getScheme());
+
+        boolean useRename = logStore.isPartialWriteVisible(targetPath, hadoopConf);
```

Comment:
local fs (which the test is using requires rename), so these changes are covered.